pkg/systemstatsmonitor/net_collector_test.go

```diff
@@ -0,0 +1,189 @@
+	netCollector.collect()
+	validate(t, netCollector)
+}
+
+func TestCollect(t *testing.T) {
+	tcs := []struct {
+		Name                   string
+		ExcludeInterfaceRegexp ssmtypes.NetStatsInterfaceRegexp
+		Validate               func(t *testing.T, nc *netCollector)
+	}{
+		{
+			Name:                   "NoFilterMatch",
+			ExcludeInterfaceRegexp: ssmtypes.NetStatsInterfaceRegexp{R: regexp.MustCompile(`^fake$`)},
+			Validate: func(t *testing.T, nc *netCollector) {
+				// We just validate two metrics, no need to check all of them
+				expectedValues := map[metrics.MetricID]map[string]int64{
+					metrics.NetDevRxBytes: map[string]int64{
+						"eth0":    5000,
+						"docker0": 1000,
+						"docker1": 500,
+						"docker2": 0,
+					},
+					metrics.NetDevTxBytes: map[string]int64{
+						"eth0":    2500,
+						"docker0": 0,
+						"docker1": 3000,
+						"docker2": 6000,
+					},
+				}
+				for metricID, interfaceValues := range expectedValues {
+					collector, ok := nc.recorder.collectors[metricID]
+					if !ok {
+						t.Errorf("Failed to get collector of metric %s", metricID)
+						continue
+					}
+					fakeInt64Metric, ok := collector.metric.(*metrics.FakeInt64Metric)
+					if !ok {
+						t.Fatalf("Failed to convert metric %s to fakeMetric", string(metricID))
+					}
+					for _, repr := range fakeInt64Metric.ListMetrics() {
+						interfaceName, ok := repr.Labels[interfaceNameLabel]
+						if !ok {
+							t.Fatalf("Failed to get label %q for ", interfaceNameLabel)
+						}
+						expectedValue, ok := interfaceValues[interfaceName]
+						if !ok {
+
+						}
+						if repr.Value != expectedValue {
+							t.Errorf("Mismatch in metric %q for interface %q: expected %d, got %d", metricID, interfaceName, expectedValue, repr.Value)
+						}
+					}
+				}
+			},
+		},
+		{
+			Name:                   "FilterMatch",
+			ExcludeInterfaceRegexp: ssmtypes.NetStatsInterfaceRegexp{R: regexp.MustCompile(`docker\d+`)},
+			Validate: func(t *testing.T, nc *netCollector) {
+				// We just validate two metrics, no need to check all of them
+				expectedValues := map[metrics.MetricID]map[string]int64{
+					metrics.NetDevRxBytes: map[string]int64{
+						"eth0": 5000,
+					},
+					metrics.NetDevTxBytes: map[string]int64{
+						"eth0": 2500,
+					},
+				}
+				for metricID, interfaceValues := range expectedValues {
+					collector, ok := nc.recorder.collectors[metricID]
+					if !ok {
+						t.Errorf("Failed to get collector of metric %s", metricID)
+						continue
+					}
+					fakeInt64Metric, ok := collector.metric.(*metrics.FakeInt64Metric)
+					if !ok {
+						t.Fatalf("Failed to convert metric %s to fakeMetric", string(metricID))
+					}
+					for _, repr := range fakeInt64Metric.ListMetrics() {
+						interfaceName, ok := repr.Labels[interfaceNameLabel]
+						if !ok {
+							t.Fatalf("Failed to get label %q for ", interfaceNameLabel)
+						}
+						expectedValue, ok := interfaceValues[interfaceName]
+						if !ok {
+
```

Comment:
t.Fatal?

Reply:
Good catch! It looks like I forgot to update this. Fixed.